src/utils/convertMarkdown.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { Fragment } from "react";
+
+import ReactMarkdown from "react-markdown";
+import glm from "remark-gfm";
+import emoji from "remark-emoji";
+import externalLinks from "remark-external-links";
+import sanitize from "rehype-sanitize";
+
+export const convertMarkdown = (
+  text: string | undefined,
+  options?: { images?: boolean; allowP?: boolean }
+) => {
+  if (!text) return;
+  const disallowed: Array<string> = [];
+  if (options?.images === false) disallowed.push("img");
+  return text.split("\\n").map((word, index) => {
+    return (
+      <ReactMarkdown
+        linkTarget={"_blank"}
+        remarkPlugins={[[glm], [emoji], [externalLinks]]}
+        rehypePlugins={[[sanitize]]}
+        key={index}
+        disallowedElements={disallowed}
+        components={options?.allowP ? {} : { p: Fragment }}
+      >
+        {word}
+      </ReactMarkdown>
+    );
+  });
+};
```

Comment:
As I believe I called out in my last review (and seemingly in https://github.com/sparkletown/sparkle/pull/1383#issuecomment-844729468), instead of this being a function in `utils/*`, it should probably be a component on `components/organisms/*`, called `RenderMarkdown`.

Reply:
ignorant in TS me will try to accomplish that now, and will report back